MACHINE-LEARNING-SERVICE/src/models/cocoSsd.model.js

```
@@ -27,25 +28,31 @@ class CocoSsd extends ObjectDetection {
 
   // Allows to load the model and decode the image in order to make a detection of the desired object.
   async predict() {
-    const model = await cocoSsd.load();
-    const imagesArray = await fs.readdir(this.pathFile);
+    try {
+      const model = await cocoSsd.load();
+      const imagesArray = await fs.readdir(this.pathFile);
 
-    const imagesToPredictArray = await Promise.all(
-      imagesArray.map(async (fileName) => {
-        const img = await fs.readFile(`${this.pathFile}${fileName}`);
-        const imgTensor = tf.node.decodeImage(new Uint8Array(img), 3);
-        const predict = await model.detect(imgTensor);
-        const data = { predict, fileName };
-        return data;
-      })
-    );
+      const imagesToPredictArray = await Promise.all(
+        imagesArray.map(async (fileName) => {
+          const img = await fs.readFile(`${this.pathFile}${fileName}`);
+          const imgTensor = tf.node.decodeImage(new Uint8Array(img), 3);
+          const predict = await model.detect(imgTensor);
+          const data = { predict, fileName };
+          return data;
+        })
+      ).catch(error =>{
+        throw new MachineLearningException('Error building model COCO.', 'ML-01');
+      });
 
-    const foundObjectsArray = FilterResults.filterFunction(
-      imagesToPredictArray,
-      this.objectRequired,
-      this.percentage
-    );
-    return foundObjectsArray;
+      const foundObjectsArray = FilterResults.filterFunction(
+        imagesToPredictArray,
+        this.objectRequired,
+        this.percentage
+      );
+      return foundObjectsArray;
+    } catch (error) {
+      throw error;
```

Should throw a message error